app/services/create_requested_account.rb

```diff
@@ -36,38 +37,55 @@ def process_request
   # such as an intermediate CAPTCHA step, we just fail immediately; users are expected
   # to have account creator rights on the target wiki, so that CAPTCHA is not required.
   def handle_mediawiki_response
-    response_status = @response.dig('createaccount', 'status')
-    if response_status == 'PASS'
-      @result = { success: "Created account for #{@username} on #{@wiki.base_url}.
-                            A password will be emailed to #{@email}." }
+    response = @response['createaccount'] || {}
+    status, message, messagecode = response.values_at('status',
+                                                      'message',
+                                                      'messagecode')
+
+    if status == 'PASS'
       create_account
-      @requested_account.destroy
-    elsif response_status == 'FAIL'
-      @result = { failure: "Could not create account for #{@username} / #{@email}.
-                            #{@wiki.base_url} message:
-                            #{@response.dig('createaccount', 'messagecode')}
-                            — #{@response.dig('createaccount', 'message')}" }
-      destroy_request_if_invalid
+    elsif status == 'FAIL' && messagecode == 'userexists'
+      destroy_request_if_user_exists(message, messagecode)
     else
-      @result = { failure: "Could not create account for #{@username} /
-                            #{@email}. #{@wiki.base_url} response: #{@response}" }
-      log_unexpected_response
+      retry_request_with_backup_account
     end
   end
 
   def create_account
+    @result = { success: "Created account for #{@username} on #{@wiki.base_url}.
+                            A password will be emailed to #{@email}." }
+
     returned_username = @response.dig('createaccount', 'username')
     raise AccountCreationError, 'no username returned' if returned_username.blank?
     @user = UserImporter.new_from_username(returned_username, @wiki)
     raise AccountCreationError, "could not create user #{returned_username}" if @user.blank?
+
+    @requested_account.destroy
   end
 
-  def destroy_request_if_invalid
+  def destroy_request_if_user_exists(message, messagecode)
+    @result = { failure: "Could not create account for #{@username} / #{@email}.
+                            #{@wiki.base_url} message:
+                            #{messagecode} — #{message}" }
+
     code = @response.dig('createaccount', 'messagecode')
     @requested_account.destroy if code == 'userexists'
   end
 
+  def retry_request_with_backup_account
+    backup_account_id = ENV['account_creation_backup_creator_id']
+    backup_account = User.find_by(id: backup_account_id)
+    return log_unexpected_response if !backup_account.is_a?(User) || backup_account == @creator
+
+    @creator = backup_account
+    creation_reason = "Created #{@username} at the request of #{@creator}."
+    process_request(backup_account, creation_reason)
```

Comment:
This will loop if the retry is also failing.

Reply:
I think the line above setting `@creator = backup_account` should solve this? On a second run through, the backup_account and the @creator should be the same.

Reply:
What happens if trying with the backup account fails (for example, because the account's credentials expired or it got blocked)? I don't see how to avoids an infinite loop in that circumstance.